### PR TITLE
Added various error checks in MissionParse and External.

### PR DESCRIPTION
**Pull request by JamesMichaelMatthews**
Added checks for xml parsing errors in MissionParse. However, note that sometimes it hard crashes, sometimes not.
Fixed sub_cb return that was getting bad data from a bad lambda.
Added error checking in the External step function, to catch the case where scrimmage was not setup properly. Also found a missing mutex unlock.

### CODE REVIEW

File: src/parse/MissionParse.cpp [hunk 1]
```diff
@@ -84,7 +84,19 @@ bool MissionParse::parse(const std::string &filename) {
...
+    //doc.parse requires a null terminated string
+    std::vector<char> contentvec(content.size() + 1); //allocation done here
+    contentvec.assign(content.begin(), content.end()); //copy
+    contentvec.push_back('\0'); //shouldn't reallocate
+    try {
+      //Note: This parse function can hard fail (seg fault, no exception) on
+      //      badly formatted xml data. Sometimes it'll except, sometimes not.
+      doc.parse<0>(contentvec.data());
+    } catch (...) {
+      cout << "scrimmage::MissionParse::parse: Exception during rapidxml::xml_document<>.parse<>()." << endl;
```
Comment (esquires) [inline src/parse/MissionParse.cpp]: Can we throw an error here instead of using cout?

File: include/scrimmage/entity/External.h [hunk 2]
```diff
@@ -178,7 +178,8 @@ class External {
         boost::optional<std::list<NetworkDevicePtr>> subs =
             pubsub_->find_subs(network_name, topic_name);
 
-        auto no_subs_msg = [&]() {
+        auto no_subs_msg = [=]() { //capture by copy, used in returned function.
+            //Note: Find_subs already prints this information. Keep it here or there?
```
Comment (esquires) [inline include/scrimmage/entity/External.h]: If it is already printed in find_subs, can we just make this function empty?

File: src/entity/External.cpp [hunk 3]
```diff
@@ -255,7 +255,7 @@ bool External::step(double t) {
     last_t_ = t;
     mutex.unlock();
 
-    this->call_update_contacts(t);
+    if(!this->call_update_contacts(t)) return false;
```
Comment (esquires) [inline src/entity/External.cpp]: This is probably a cpplint error.

File: src/parse/MissionParse.cpp (hunk 1 above)
Comment (JamesMichaelMatthews) [reply src/parse/MissionParse.cpp]: It is already. It returns false, that's the error mechanism used here.

File: include/scrimmage/entity/External.h (hunk 2 above)
Comment (JamesMichaelMatthews) [reply include/scrimmage/entity/External.h]: Idk. I would, but it wasn't like that before, so you decide. Maybe you'd prefer to remove it from the other location, how'd you want to design it. You wrote this didn't you?

File: src/entity/External.cpp (hunk 3 above)
Comment (JamesMichaelMatthews) [reply src/entity/External.cpp]: Looks good to me. What's the problem?

**Comment by JamesMichaelMatthews** [comment]
I had to rebase again. Initially I didn't realize I rebased onto origin/master but didn't fetch that remote. Fixed the conflicts.

**Comment by esquires** [comment]
Thanks!

**Pull request merged**